### PR TITLE
Enable OS X builds on Jenkins

### PR DESCRIPTION
**Pull request by digitalinfinity**
This change enables OS X build/test as part of Jenkins CI. Ironically, this is currently targetted only to the linux branch but we'll bring it back to master soon.
Changes:
- Update groovy script to generate OS X build tasks
- Update get_system_info.sh to work on OS X
- Disable one failing test on OS X in debug test runs

### CODE REVIEW

**Comment by digitalinfinity** [comment]
@dilijev please review

File: netci.groovy [hunk 1]
```diff
@@ -113,11 +114,19 @@ def CreateBuildTasks = { machine, configTag, buildExtra, testExtra, runCodeAnaly
...
-                def config = "linux_${buildType}"
+            def staticBuildConfigs = [true, false]
+            if (platform == "osx") {
+                staticBuildConfigs = [true]
+            }
+
+            staticBuildConfigs.each { staticBuild ->
+                def config = (platform == "osx" ? "osx_${buildType}" : "linux_${buildType}")
+                def numConcurrentCommand = (platform == "osx" ? "sysctl -n hw.logicalcpu" : "nproc")
+
```
Comment (dilijev) [inline netci.groovy]: nit trailing spaces

File: netci.groovy [hunk 2]
```diff
@@ -113,11 +114,19 @@ def CreateBuildTasks = { machine, configTag, buildExtra, testExtra, runCodeAnaly
...
     [true, false].each { isPR ->
         ['debug', 'test', 'release'].each { buildType ->
-            [true, false].each { staticBuild ->
-                def config = "linux_${buildType}"
+            def staticBuildConfigs = [true, false]
+            if (platform == "osx") {
+                staticBuildConfigs = [true]
+            }
+
+            staticBuildConfigs.each { staticBuild ->
```
Comment (dilijev) [inline netci.groovy]: Nice!

File: netci.groovy [hunk 3]
```diff
@@ -113,11 +114,19 @@ def CreateBuildTasks = { machine, configTag, buildExtra, testExtra, runCodeAnaly
...
 
-def CreateLinuxBuildTasks = { machine, configTag, linuxBranch, nonDefaultTaskSetup ->
+// Generic task to trigger clang-based cross-plat build tasks
+def CreateXPlatBuildTasks = { machine, platform, configTag, xplatBranch, nonDefaultTaskSetup ->
     [true, false].each { isPR ->
         ['debug', 'test', 'release'].each { buildType ->
-            [true, false].each { staticBuild ->
-                def config = "linux_${buildType}"
+            def staticBuildConfigs = [true, false]
+            if (platform == "osx") {
```
Comment (dilijev) [inline netci.groovy]: >osx [](start = 29, length = 3)

Should all instances of "osx" be "OSX"?

File: test/Error/rlexe.xml [hunk 4]
```diff
@@ -30,6 +30,8 @@
     <default>
       <files>stack.js</files>
       <baseline>stack.baseline</baseline>
+      <!-- xplat-todo: Understand why this test is failing on OS X -->
+      <tags>exclude_mac</tags>
```
Comment (dilijev) [inline test/Error/rlexe.xml]: >exclude_mac [](start = 12, length = 11)

Why not "exclude_osx"?

**Comment by dilijev** [comment]
:shipit:

File: test/Error/rlexe.xml (hunk 4 above)
Comment (obastemur) [inline test/Error/rlexe.xml]: @digitalinfinity this should not fail. Can you share the log/output please?

Comment (digitalinfinity) [inline test/Error/rlexe.xml]: Hmm, interesting- it had failed in my test PR which is why I had disabled it, but I just reran the tests on that PR and that test passed. However, another test failed now, which worries me. @obastemur - are the tests on OS X stable, or do we need to temporarily disable tests for that platform until we improve the stability? See #1329 for logs for the current failure. I'm not on corpnet right now, I'll try to investigate more once I'm in the office.

Comment (digitalinfinity) [inline test/Error/rlexe.xml]: Direct link to the logs: http://dotnet-ci.cloudapp.net/job/Microsoft_ChakraCore/job/linux-ci/job/osx_osx_debug_static_prtest/10/

Comment (obastemur) [inline test/Error/rlexe.xml]: That object inline test fail also first for me. Can we have some env. detail on osx CI? we might also need to change test timeouts

Comment (obastemur) [inline test/Error/rlexe.xml]: Couldn't reproduce yet. Created a loop script running tests all day long..

Comment (obastemur) [inline test/Error/rlexe.xml]: Fix for this random issue #1339

Comment (digitalinfinity) [inline test/Error/rlexe.xml]: Cool, thanks @obastemur- I'll rebase this change and remove the exclude_mac bit once you've merged your change in.

Comment (digitalinfinity) [inline test/Error/rlexe.xml]: Alternatively, I could merge my change in, you could rebase on top of this and use it to verify that the Jenkins PR is working- what do you prefer?

File: netci.groovy (hunk 3 above)
Comment (digitalinfinity) [inline netci.groovy]: No :( osx = platform (that's our internal parameter), OSX = Jenkins parameter. I had to separate them out for consistency with Linux 😞

**Comment by digitalinfinity** [comment]
Thanks @dilijev. Merging.

File: netci.groovy (hunk 3 above)
Comment (dilijev) [inline netci.groovy]: Yeah, unfortunate but it actually looks better this way. This is fine.

**Pull request merged**